src/wallet/wallet2.cpp

```diff
@@ -6126,6 +6126,19 @@ std::map<uint32_t, uint64_t> wallet2::balance_per_subaddress(uint32_t index_majo
         amount_per_subaddr[0] = utx.second.m_change;
       else
         found->second += utx.second.m_change;
+
+      // add transfers to same wallet
+      for (const auto &dest: utx.second.m_dests) {
```

Comment:
Does anyone know a better way than using destinations?

It may not be ideal or as efficient to reverse lookup from a destination address. (?)

Also, destination addresses are only present in the wallet cache, but this change only affects unconfirmed funds so the balance becomes correct when the funds are confirmed.